### DISCUSSION TITLE
Single Antenna Operation with srsRAN gNB on 5Gnr TDD bands

### DISCUSSION THREAD

**Discussion by alex-arknetworx** [srsran/srsRAN_Project#195] [General Help]
Are there any srsRAN community members that might have experimented with the use of srsRAN gNB in conjunction with an RF Circulator in order to multiplex the TX and RX chains of the USRP onto the same antenna when operating on a TDD band?

Much like the use of a Duplexer (e.g. Sysmocom's range of cavity duplexers) when operating on FDD bands? Is this even possible or would there also be a need for an RF Switch (along with a switching output signal from the gNB software) to isolate the RX chain to prevent overloading it during TX'ing?

Very interested in any circulator part numbers and if any members have tried srsRAN Project with a Power Amplifier on the TX or a Low Noise Amplifier on the RX as well.

**Comment by fllay** [comment]
I plan to use this one.
https://th.mouser.com/ProductDetail/Molex/73591-2049?qs=Znm5pLBrcAJXiRvgixSH5Q%3D%3D

For RF frontend, I have heard that this one is good but I still does not contact them for pricing.

http://zhixun-wireless.top/product

**Reply by robertfalkenberg** [reply]
This video "building wired RF test setups for cellular" [1] could have some useful infos as a starting point, especially regarding the maximum input levels of the expensive devices.

_Disclaimer: Please ensure sufficient attenuation. No responsibility will be taken for bricked equipment._

[1] https://osmocom.org/projects/osmo-dev-con/wiki/OsmoDevCall

**Reply by fllay** [reply]
Is it possible to expose RX signal to b210 GPIO since RX LED is not expose to any pin? If so, we can use something like this https://www.qorvo.com/feature/reference-design-mmimo-rffe#related-products and we can connect RX signal to the enable pin of the LNA. In this case, we will have a lot of isolation since the LNA is in disable state.

**Reply by aldalahmeh** [reply]
That would be interesting, I hope it is possible.

**Reply by pgawlowicz** [reply]
@fllay @aldalahmeh there is this nice presentation: https://events.gnuradio.org/event/18/contributions/234/attachments/74/186/GPIOs%20on%20USRPs.pdf

**Comment by alex-arknetworx** [comment]
@fllay thanks for sharing, particularly the zhixun wireless product link: there are a bunch of interesting products which may be suitable, I will work my way through the specifications. @robertfalkenberg i will have a look at the video as yes my initial thoughts were that just using a circulator may not offer enough isolation loss between the tx and rx chains, especially if using a PA thus overloading or damaging the usrp.

I'm assuming the gNB software itself rejects received samples during it's transmission timeslot in TDD mode (using the sample timestamps?) as even with two antennas directly connected to USRP (and oriented 90 degrees to each other), the gNB would be 'hearing' it's own TX?

**Reply by aldalahmeh** [reply]
Hello,
I am interested in this topic as well. Did you manage to come up with a solution?
